### ISSUE TITLE
Resources for learning and as reference for ROS2, NN, etc.

### ISSUE THREAD

**Issue by Namnodorel** [felar/robug#1]
# Python
- [PyCharm Education: An IDE with simple courses showing the basics of using Python](https://www.jetbrains.com/pycharm-edu/)
- [Python 3 Standard Library Reference](https://docs.python.org/3/library/index.html)

# Turtlebot3 and ROS2
## Setup
- [Setting up Turtlebot3 with ROS2](http://emanual.robotis.com/docs/en/platform/turtlebot3/ros2) _(OS used is Ubuntu 18.04 Bionic)_
- [Building ROS2 from source](https://index.ros.org/doc/ros2/Installation/Linux-Development-Setup/) _(Necessary because the latest ROS2 release doesn't include some bugfixes that are required for TB3 to work)_

## Usage
- [ROS2 concepts document](https://index.ros.org/doc/ros2/Concepts/Overview-of-ROS-2-concepts/)
- [Topics, Publishers and Subscribers](https://www.youtube.com/watch?v=bJB9tv4ThV4)_(The tutorial is about ROS1, but still accurate for ROS2 until **3:06**)_
- [`rclpy` API reference](http://docs.ros2.org/crystal/api/rclpy/index.html)_(`rclpy` is the Python API for accessing ROS2)_
- [Minimalistic examples of using different features of `rclpy` ](https://github.com/ros2/examples/tree/master/rclpy)
- [Awesome ROS2](https://fkromer.github.io/awesome-ros2/) _(A list of existing ROS2 libraries and resources that may be useful)_
- [ROS2 and Tensorflow image detection example](https://github.com/alsora/ros2-tensorflow) _(**Outdated**, both regarding ROS2 and Tensorflow. May still be useful as an additional example to learn from)_

# Neural Networks
## General Neural Networks
- [A general introduction to neural networks, gradient descent and backpropagation ](https://www.youtube.com/watch?v=aircAruvnKk&list=PLLMP7TazTxHrgVk7w1EKpLBIDoC50QrPS)_(by 3Blue1Brown)_
- [Deep Learning with Tensorflow](https://www.youtube.com/watch?v=vq2nnJ4g6N0)_(by Martin Görner)_
- [Tensorflow 2.0 Beginner Tutorials](https://www.tensorflow.org/alpha/tutorials/keras)
- [All lectures by Martin Görner](https://github.com/GoogleCloudPlatform/tensorflow-without-a-phd)_(long but very informative and well explained)_

## Reinforcement Learning
- [Introduction to Reinforcement Learning](https://www.youtube.com/watch?v=t1A3NTttvBA)_(by Martin Görner)_
- [Reinforcement Learning with Tensorflow-Agents](https://youtube.com/watch?v=tAOApRQAgpc)_(by Sergio Guadarrama and Eugene Brevdo)_
- [Example of Reinforcement Learning using Tensorflow and ROS to make the TB3 avoid obstacles and reach a goal](http://emanual.robotis.com/docs/en/platform/turtlebot3/machine_learning/#machine-learning) [_(Implementation)_](https://github.com/ROBOTIS-GIT/turtlebot3_machine_learning/blob/master/turtlebot3_dqn/src/turtlebot3_dqn/environment_stage_1.py) [ ! ]

- [TF-Agents Environments example](https://github.com/tensorflow/agents/blob/master/tf_agents/colabs/2_environments_tutorial.ipynb) 
- [TF-Agents Policies example](https://github.com/tensorflow/agents/blob/master/tf_agents/colabs/3_policies_tutorial.ipynb) 
- [TF-Agents Drivers example](https://github.com/tensorflow/agents/blob/master/tf_agents/colabs/4_drivers_tutorial.ipynb) 
- [TF-Agents Replay Buffer example](https://github.com/tensorflow/agents/blob/master/tf_agents/colabs/5_replay_buffers_tutorial.ipynb) 
- [Reinforcement Learning with TF-Agents for a cartpole example](https://colab.research.google.com/github/tensorflow/agents/blob/master/tf_agents/colabs/6_reinforce_tutorial.ipynb)

# Writing Documentation
- [Creating pages and adding them to the table of contents](https://docsify.js.org/#/more-pages)
- [Markdown Cheat Sheet/overview](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
- [Complete markdown reference](https://github.github.com/gfm/)  
- [How to use mermaid (to create graphs)](https://mermaidjs.github.io/#/flowchart)
- [Preview mermaid graphs](https://mermaidjs.github.io/mermaid-live-editor/)
- mermaid graphs need to be wrapped in ```mermaid blocks
- [LaTeX math expression editor](https://www.zahlen-kern.de/editor/)
- LaTeX expressions need to be wrapped in ```mermaid blocks
